src/Core/Domain/Store/ValueObject/Coordinate.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Core\Domain\Store\ValueObject;
+
+use PrestaShop\PrestaShop\Core\ConstraintValidator\Constraints\TypedRegex;
+use PrestaShop\PrestaShop\Core\Domain\Exception\DomainConstraintException;
+use PrestaShop\PrestaShop\Core\Domain\ValueObject\ValueObject;
+use Symfony\Component\Validator\Constraints\NotBlank;
+
+/**
+ * Class Coordinate is responsible for providing valid coordinate value.
+ */
+class Coordinate extends ValueObject
+{
+    /**
+     * @var float
+     */
+    private $coordinate;
+
+    /**
+     * @param string $coordinate
+     *
+     * @throws DomainConstraintException
+     */
+    public function __construct($coordinate)
+    {
+        $this->validate($coordinate, [
+            new NotBlank(),
+            new TypedRegex('coordinate'),
```

Comment:
```suggestion
            new TypedRegex(['type' => 'coordinate']),
```

Reply:
@tom-combet Could you add these modifications ?

Reply:
Are you sure that this is necessary? Type is the default option so it can be passed directly without specifying it in an array.
So it should be functional like this, but maybe it is for clarity? And if so, I'll make the changes ofc.